wagtailmenus/models/menuitems.py

```diff
@@ -91,7 +91,11 @@ def menu_text(self):
 
     def relative_url(self, site=None):
         if self.link_page:
-            return self.link_page.relative_url(site) + self.url_append
```

Comment:
I'd be happier just changing this line to

`return self.link_page.relative_url(site) or '' + self.url_append`